test/optimizer/rule_test.cpp

```diff
@@ -54,5 +69,176 @@ TEST_F(RuleTests, SimpleRuleApplyTest) {
   EXPECT_EQ(outputs.size(), 1);
 }
 
+// Test whether update stament will use index scan plan
+TEST_F(RuleTests, UpdateDelWithIndexScanTest) {
```

Comment:
This test relies on a functional parser and planner, it is more like a integrate test instead of a unit test. When the test fails it would be hard to determine which part is wrong. To make it more straightforward and clearer, you can consider following this pattern:

1. Prepare the test table using code instead of SQL
2. Construct the test plan with code, see rule_test.cpp#SimpleRuleApplyTest
3. Feed the constructed plan directly to simple optimizer
4. Check the output plan of simple optimizer

For how to create and load a table using code, please have a look at `executor_tests_util.cpp` (and yes it would be great if we also add ab utility class for optimizer, given that we don't actually have a test for optimizer, it will be a good start)